### PR TITLE
[REF]  minor refactor of get_cmap and matplotlib version-handling

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but alternative to https://github.com/nilearn/nilearn/pull/4419

Changes proposed in this pull request:

- keep the few refactors that could be salvaged from #4419

### CODE REVIEW

File: nilearn/plotting/img_plotting.py
Comment (Remi-Gau) [inline nilearn/plotting/img_plotting.py:1]: mostly simplify imports

File: nilearn/plotting/cm.py [hunk 1]
```diff
@@ -291,16 +296,16 @@ def _revcmap(data):
 globals().update(_cmap_d)
 # Register cmaps in matplotlib too
 for k, v in _cmap_d.items():
-    try:
+    if compare_version(matplolib.__version__, ">=", "3.5.0"):
```
Comment (Remi-Gau) [inline nilearn/plotting/cm.py:299]: make the code comment explicit

File: nilearn/plotting/tests/test_img_plotting/test_plot_roi.py [hunk 2]
```diff
@@ -84,10 +84,6 @@ def test_cmap_with_one_level(shape_3d_default, affine_eye):
 
     clust_ids = list(np.unique(img.get_fdata())[1:])
 
-    try:
-        cmap = plt.colormaps["tab20"].resampled(len(clust_ids))
-    except TypeError:
-        # for older versions of matplotlib
```
Comment (Remi-Gau) [inline nilearn/plotting/tests/test_img_plotting/test_plot_roi.py:90]: does not seem to be needed

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Review by emdupre — APPROVED** [review]
This LGTM, thank you @Remi-Gau  !

**Pull request merged**